File: backend/GestionCommerciale.Api/Models/Order.cs
namespace GestionCommerciale.Api.Models;

public class Order
{
    public int Id { get; set; }
    public string NumeroCommande { get; set; } = string.Empty;
    public int ClientId { get; set; }
    public Client? Client { get; set; }
    public DateTime DateCommande { get; set; } = DateTime.UtcNow;
    public OrderStatus Statut { get; set; } = OrderStatus.Brouillon;
    public decimal TotalHT { get; set; }
    public decimal TotalTTC { get; set; }

    public ICollection<OrderLine> OrderLines { get; set; } = new List<OrderLine>();
}

File: frontend/src/models/order.ts
export interface OrderLine {
  id: number;
  productId: number;
  productNom: string;
  quantite: number;
  prixUnitaire: number;
  totalLigne: number;
}

export interface Order {
  id: number;
  numeroCommande: string;
  clientId: number;
  clientNom: string;
  dateCommande: string;
  statut: string;
  totalHT: number;
  totalTTC: number;
  lignes: OrderLine[];
}

export interface CreateOrderLineDto {
  productId: number;
  quantite: number;
}

export interface CreateOrderDto {
  clientId: number;
  lignes: CreateOrderLineDto[];
}